### PR TITLE
itwinjs 3.0 upgrade

### PR DESCRIPTION
**Pull request by aruniverse**
Should be able to simplify it a bit further on the viewer side, with the useAccessToken hook, but it wasnt updating like i expected, need to debug that in the viewer repo.

Nuked the proxy server, everyone should use the token server approach. Need to add the `imodels:read` scope now (required by the viewer)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,37 +6,19 @@ There are two distinct ways of handling user identities when building an applica
 
 ## Client Registration
 
-The first step in either implementation is registering a new client id as a "Service" type in the [iTwin App Registration](https://developer.bentley.com/my-apps/) and add the 'Visualization' API Association. This will provide a client id and client secret to use for getting an access token, via a [client credentials workflow](https://developer.bentley.com/apis/overview/authorization/#clientcredentialflow), to access the iTwin Platform.
+The first step in either implementation is registering a new client id as a "Service" type in the [iTwin App Registration](https://developer.bentley.com/my-apps/) and add the `Visualization` and `iModels` API Association. This will provide a client id and client secret to use for getting an access token, via a [client credentials workflow](https://developer.bentley.com/apis/overview/authorization/#clientcredentialflow), to access the iTwin Platform.
```
Comment (calebmshafer) [inline README.md]: Let's reword this the first line to "The first step is to register a new client is as a ...". Since we're removing the other way you could set this up.

File: README.md [hunk 2]
```diff
@@ -6,37 +6,19 @@ There are two distinct ways of handling user identities when building an applica
...
 
 > Don't forget to add `<CLIENT_ID>@apps.imsoidc.bentley.com` to your Project too!
 
 The client secret provided when creating the client is intended to be kept __secret__ and not used client-side. Therefore it is __strongly__ recommended to follow one of the two approaches to build a server-side component that manages the secret and avoid having to cache or bundle the secret into your web app. The server-side will hold the client secret created for your application and manages using the token.
```
Comment (calebmshafer) [inline README.md]: "recommend to build a server-side component that manages the secret..."

File: README.md [hunk 3]
```diff
@@ -6,37 +6,19 @@ There are two distinct ways of handling user identities when building an applica
...
 
-## Server Side Implementations
+## Server Side Implementation
 
 > __Important__: It is strongly recommended that any server-side component is placed behind a layer of authorization and/or authentication that fits your workflow and validates the user has access to what is being requested. However to simplify the example, this server will not be setup with any additional checking and essentially provide an "unprotected" endpoint.
 
-The two different implementations;
 
-1. The first approach is to use a proxy server between the end user application (e.g. iTwin Viewer) and the iTwin Platform to forward all requests while swapping out the authentication header when sent to the platform. The proxy server is in charge of taking the incoming request and updating the authentication header with an access token that is generated using the client credentials workflow mentions above.
+The recommended approach is to use a token server to hold the client id and client secret and provides an endpoint which can be called by an Auth Client for the iTwin Viewer to get the jwt token required to call the iTwin Platform.
```
Comment (calebmshafer) [inline README.md]: nit: Auth Client can be lower cased

File: react-viewer/README.md [hunk 4]
```diff
@@ -4,11 +4,11 @@ This project was bootstrapped with [Create React App](https://github.com/faceboo
 
 ## Environment Variables
 
-Prior to running the app, add a valid contextId, iModelId, changeSetId for your user in the .env file:
+Prior to running the app, add a valid iTwinId, iModelId, changeSetId for your user in the .env file:
```
Comment (calebmshafer) [inline react-viewer/README.md]: nit: changeSetId -> changesetId

File: react-viewer/tsconfig.json [hunk 5]
```diff
@@ -13,8 +17,10 @@
     "resolveJsonModule": true,
     "isolatedModules": true,
     "noEmit": true,
-    "jsx": "react",
+    "jsx": "react-jsx",
```
Comment (calebmshafer) [inline react-viewer/tsconfig.json:20]: Is this change needed?

File: token-server/yarn.lock [hunk 6]
```diff
@@ -0,0 +1,1251 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
+"@itwin/core-bentley@rc":
```
Comment (calebmshafer) [inline token-server/yarn.lock]: Can we not use yarn? :)

File: react-viewer/tsconfig.json (hunk 5 above)
Comment (aruniverse) [reply react-viewer/tsconfig.json:20]: default change automatically made by cra when you use react 17; should be negligible

Comment (calebmshafer) [reply react-viewer/tsconfig.json:20]: okay, didn't know we changed that in the core. Fine by me

File: react-viewer/package.json [hunk 7]
```diff
@@ -1,43 +1,48 @@
...
-    "@bentley/ui-abstract": "^2.19.1",
-    "@bentley/ui-components": "^2.19.1",
-    "@bentley/ui-core": "^2.19.1",
-    "@bentley/ui-framework": "^2.19.1",
-    "@bentley/ui-ninezone": "^2.19.1",
-    "@bentley/webgl-compatibility": "^2.19.1",
-    "@itwin/web-viewer-react": "^1.1.0",
-    "react": "^16.13.1",
-    "react-dom": "^16.13.1"
+    "@bentley/icons-generic": "^1.0.13",
```
Comment (calebmshafer) [inline react-viewer/package.json:6]: Can we remove this yet? It's been something I've been wondering about the base viewer template :)

Comment (aruniverse) [reply react-viewer/package.json:6]: unfortunately not :(
a lot of icons used in core and other pkgs require this pkg to be present to deliver the webfonts needed for icons
i really want to nuke this pkg in 4.0

Comment (calebmshafer) [reply react-viewer/package.json:6]: that's annoying okay, I'll add it to the 4.0 list.

**Review by calebmshafer — APPROVED** [review]
LGTM! Just tested it out and works well. Thanks for picking this up!

Side note: I guess I should rename the repo at some point...
